includes/class-wc-stripe-helper.php

```diff
@@ -577,7 +577,7 @@ public static function get_dynamic_statement_descriptor_suffix( $order ) {
 
 			// Stripe requires at least 1 latin (alphabet) character in the suffix so we add the first character of the prefix before the order number.
 			if ( 0 === preg_match( '/[a-zA-Z]/', $suffix ) ) {
-				$suffix = substr( $prefix, 0, 1 ) . ' ' . $suffix;
+				$suffix = ! empty( $prefix ) ? substr( $prefix, 0, 1 ) . ' ' . $suffix : 'O ' . $suffix;
```

Comment:
Added an `O` before the order id when the prefix is empty. This was originally thought of as a solution in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2833/files#r1453081172

Reply:
How about using `ID` instead of `O`? I think it's easier to relate the order number with an ID, rather than try to guess what the `O` is about.